src/portfolios/portfolio/components/TaskOrder/TaskOrderDetails.vue

```diff
@@ -169,10 +395,211 @@ export default class TaskOrderDetails extends Vue {
   get isAlertClosed(): boolean {
     return AcquisitionPackage.taskOrderDetailsAlertClosed
   }
+
+  public getValign(item: ClinTableRowData): string {
+    return item.isPending || (item.isExpired && !item.isOverspent) ? "middle" : "top"
+  }
+
   public handleClick(): void {
     this._showDetails = false
   }
 
+  public toCurrencyString(value: number): string {
+    return toCurrencyString(value)
+  }
+
+  public tableHeaders: Record<string, string>[] = [
+    { text: "CLIN", value: "CLINNumber" },
+    { text: "Status", value: "status" },
+    { text: "Period of performance", value: "PoP" },
+    { text: "Total CLIN value", value: "totalCLINValue" },
+    { text: "Obligated funds", value: "obligatedFunds" },
+    { text: "Total funds spent (%)", value: "totalFundsSpent" },
+  ];
+ 
+  public toggleInactive():void {
+    this.showInactive = !this.showInactive
+  }
+
+  public fundsRemaining(obligatedFunds:string | number, fundsSpent:string | number):
+    { percent:string, fundsRemaining:string } {
+    if(obligatedFunds == "0" && fundsSpent == "0"){
+      return {
+        percent: "0",
+        fundsRemaining: ""
+      }
+    }
+    let percent = Math.round((Number(fundsSpent) / Number(obligatedFunds)) * 100);
+    let remaining = Number(obligatedFunds) - Number(fundsSpent);
+    return {
+      percent: String(percent),
+      fundsRemaining:"$" + toCurrencyString(remaining) + " remaining"
+    }
+  }
+
+  public timeToExpiration(start:string,end:string): Record<string, string> {
```

Comment:
Can we move this function and the respective template code into it own util component? Component that takes in "start" and "end" date properties and does the rest and displays. This change will reduce the lines in this already large component by about 60 lines and potentially any duplications of this elsewhere.

Reply:
good idea. moved to `helpers/index.ts`